denitrip/angular-shop-backend
# task-7: added auth service configuration

## Score

5/6

## Link to the task

[click](https://github.com/rolling-scopes-school/cloud-development-course-initial/blob/main/7_authorization/task.md)

## Link to frontend deployment

[click](https://d3ik4nah6a0lnv.cloudfront.net/)

## Link to frontend PR

[click](https://github.com/denitrip/shop-angular-cloudfront/pull/5)

## Description

1 - authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file
3 - Import Service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.
5 - Client application is updated to send "Authorization: Basic authorization_token" header on import. Client should get authorization_token value from browser [localStorage](https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage)

## Demo

# with token
![with_token](https://user-images.githubusercontent.com/29579085/196879015-f3f56b8c-e887-4cc6-9045-561c39bf49e1.png)

# first without token and then with a wrong token

![image](https://user-images.githubusercontent.com/29579085/196896403-78e61502-e08e-42e8-9ca3-fc41196e54e3.png)


## demo file

[productsFileBig.csv](https://github.com/denitrip/angular-shop-backend/files/9827161/productsFileBig.csv)

## How to test

1. please open frontend deployment in incognito mode
2. try to import file without providing a token (an error message should be shown)
3. reload the page using `ctrl` + `F5`
4. add token to local storage using the command `localStorage.setItem('authorization_token','ZGVuaXRyaXA6VEVTVF9QQVNTV09SRA==')`
5. send the file once again (no error should be shown)
